### PR TITLE
support returning run info for llms, chat models and chains

### PR DESCRIPTION
**Pull request by agola11**
returning the run id is important for accessing the run later on

### CODE REVIEW

File: langchain/chains/base.py [hunk 1]
```diff
@@ -108,6 +108,8 @@ def __call__(
         inputs: Union[Dict[str, Any], Any],
         return_only_outputs: bool = False,
         callbacks: Callbacks = None,
+        *,
+        include_run_info: bool = False,
```
Comment (nfcampos) [inline langchain/chains/base.py:112]: Why make this optional?

File: langchain/chat_models/base.py [hunk 2]
```diff
@@ -93,6 +94,8 @@ def generate(
         generations = [res.generations for res in results]
         output = LLMResult(generations=generations, llm_output=llm_output)
         run_manager.on_llm_end(output)
+        if run_manager:
+            output.run = RunInfo(run_id=run_manager.run_id)
```
Comment (nfcampos) [inline langchain/chat_models/base.py:98]: Should this be RUN_KEY

Comment (agola11) [reply langchain/chat_models/base.py:98]: unfortunately you cannot set underscore keys in pydantic objects, i.e. `LLMResult`

Comment (nfcampos) [reply langchain/chat_models/base.py:98]: Wow these python libraries are something else

File: langchain/chains/base.py (hunk 1 above)
Comment (agola11) [reply langchain/chains/base.py:112]: Mainly for better bw-compat, we have a bunch of existing tests that check the returned dict from `__call__` and expect it not to have a `__run` field. Other people might have tests like that too.

File: langchain/chat_models/base.py (hunk 2 above)
Comment (vowelparrot) [reply langchain/chat_models/base.py:98]: Could set the alias then serialize with `by_alias=True` though maybe that's even more complicated

Comment (agola11) [reply langchain/chat_models/base.py:98]: yeah... seems more complex than it needs to imo

**Pull request merged**